database/models/sensors.js

```diff
@@ -5,38 +5,67 @@ sql.setDialect('postgres');
 var databaseP = require('../management/databaseClientP');
 var getRandomName = require('pokemon-names').random;
 
-var sensors = require('../management/declarations.js').sensors;
-var sensorCache = 'RIEN';
+var sensorTable = require('../management/declarations.js').sensors;
+var outputTable = require('../management/declarations.js').outputs;
 
 module.exports = {
-    cache: function(){
-        return sensorCache;
-    },
     create: function (data) {
-        sensorCache = 'TEST';
 
+        var sensorData = {};
+        var types = [];
+
+        Object.keys(data).forEach(function(key){
+            if (key === 'outputs')
+                types = data[key];
+            else
+                sensorData[key] = data[key];     
+        });
+
+        // first create the sensor
         return databaseP.then(function (db) {
 
-            if (data.sim === undefined || (typeof(data.sim) === 'string' && !data.sim.length)) {
+            if (sensorData.sim === undefined || (typeof(sensorData.sim) === 'string' && !sensorData.sim.length)) {
                 throw 'Cannot create sensor : no SIM';
             }
-            if (!data.name)
-                data.name = getRandomName();
-            var query = sensors
-                .insert(data)
+            if (!sensorData.name)
+                sensorData.name = getRandomName();
+            var query = sensorTable
+                .insert(sensorData)
                 .returning('*')
                 .toQuery();
 
-            //console.log('sensors create query', query);
             return new Promise(function (resolve, reject) {
                 db.query(query, function (err, result) {
                     if (err) reject(err);
-                    else{
-                        sensorCache = 'QUELQUE CHOSE'; // need to ask David about that
-                        resolve(result.rows[0]);
-                    }
-                        
-                        
+                    else
+                        resolve(result.rows[0]);        
+                });
+            })
+            // then create the corresponding inputs
+            .then(function(sensor){
+
+                var outputPs = types.map(function(type){
```

Comment:
This sends as many INSERT are there are types. You can pass an array to .insert(), so that this is a single query. A single bigger query is usually preferable to lots of small ones.
(and you won't need the Promise.all below).